core/trino-main/src/test/java/io/trino/block/BlockAssertions.java

```diff
@@ -596,4 +626,237 @@ public static RunLengthEncodedBlock createRLEBlock(long value, int positionCount
+    {
+        return createLongDecimalsBlock(generateListWithNulls(positionCount, nullRate,
+                () -> Double.toString(RANDOM.nextDouble() * RANDOM.nextInt())));
+    }
+
+    public static Block createRandomLongsBlock(int positionCount, float nullRate)
+    {
+        return createLongsBlock(generateListWithNulls(positionCount, nullRate, RANDOM::nextLong));
+    }
+
+    public static Block createRandomSmallintsBlock(int positionCount, float nullRate)
+    {
+        return createTypedLongsBlock(
+                SMALLINT,
+                generateListWithNulls(positionCount, nullRate, () -> RANDOM.nextLong() % Short.MIN_VALUE));
+    }
+
+    public static Block createRandomStringBlock(int positionCount, float nullRate, int maxStringLength)
+    {
+        return createStringsBlock(
+                generateListWithNulls(positionCount, nullRate, () -> generateRandomStringWithLength(maxStringLength)));
+    }
+
+    private static String generateRandomStringWithLength(int length)
+    {
+        String symbols = "abcdefghijklmnopqrstuvwxyz";
+        char[] chars = new char[length];
+        for (int i = 0; i < length; i++) {
+            chars[i] = symbols.charAt(RANDOM.nextInt(symbols.length()));
+        }
+        return new String(chars);
+    }
+
+    public static RunLengthEncodedBlock createRleBlockWithRandomValue(Block block, int positionCount)
+    {
+        checkArgument(block.getPositionCount() > 0, format("block positions %d is less than or equal to 0", block.getPositionCount()));
+        return new RunLengthEncodedBlock(block.getRegion(block.getPositionCount() / 2, 1), positionCount);
+    }
+
+    public static Block wrapBlock(Block block, int positionCount, List<Encoding> wrappings)
+    {
+        checkArgument(!wrappings.isEmpty(), "wrappings is empty");
+
+        Block wrappedBlock = block;
+
+        for (int i = wrappings.size() - 1; i >= 0; i--) {
+            switch (wrappings.get(i)) {
+                case DICTIONARY:
+                    wrappedBlock = createRandomDictionaryBlock(wrappedBlock, positionCount, true);
+                    break;
+                case RUN_LENGTH:
+                    wrappedBlock = createRleBlockWithRandomValue(wrappedBlock, positionCount);
+                    break;
+                default:
+                    throw new IllegalArgumentException(format("wrappings %s is incorrect", wrappings));
+            }
+        }
+        return wrappedBlock;
+    }
+
+    static <T> Iterable<T> generateListWithNulls(int positionCount, float nullRate, Supplier<T> valueSupplier)
+    {
+        List<T> result = new ArrayList<>(positionCount);
+
+        Set<Integer> nullPositions = chooseNullPositions(positionCount, nullRate);
+        for (int i = 0; i < positionCount; i++) {
+            result.add(nullPositions.contains(i) ? null : valueSupplier.get());
+        }
+        return unmodifiableList(result);
```

Comment:
that should return `ImmutableList.copyOf` as we use Guava throughout the code for immutable collections. This way `copyOf` are noops in most cases